### PR TITLE
Remove ISchema in TextLoader.cs and TextLoaderCursor.cs

### PR DESCRIPTION
**Pull request by wschin**
As title. This might be the tail of #1501.

### CODE REVIEW

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs [hunk 1]
```diff
@@ -520,26 +520,25 @@ public static ColInfo Create(string name, PrimitiveType itemType, Segment[] segs
             }
         }
 
-        private sealed class Bindings : ISchema
+        private sealed class Bindings
         {
+            /// <summary>
+            /// <see cref="Infos"/>[i] stores the i-th column's name type loaded from the input text file.
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs]: >name type [](start = 64, length = 9)

Not a big deal, but what is a "name type"?

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs [hunk 2]
```diff
@@ -520,26 +520,25 @@ public static ColInfo Create(string name, PrimitiveType itemType, Segment[] segs
...
             public readonly ColInfo[] Infos;
-            public readonly Dictionary<string, int> NameToInfoIndex;
+            /// <summary>
+            /// <see cref="Infos"/>[i] stores the i-th column's metadata, slot names.
+            /// </summary>
             private readonly VBuffer<ReadOnlyMemory<char>>[] _slotNames;
-            // Empty iff either header+ not set in args, or if no header present, or upon load
-            // there was no header stored in the model.
+            /// <summary>
+            /// Empty iff either header+ not set in args, or if no header present, or upon load
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs]: >either header+ not set in args [](start = 26, length = 30)

I might prefer something like an actual reference using a `<see` tag to the `Arguments.HasHeader` field, given that we're now making this an XML comment. Not a big deal though.

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs (hunk 1 above)
Comment (wschin) [reply src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs]: `name type` is changed to `name and type`. Thanks.

---
In reply to: [247681984](https://github.com/dotnet/machinelearning/pull/2140#discussion_r247681984) [](ancestors = 247681984)

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs (hunk 2 above)
Comment (wschin) [reply src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs]: Now we have
```charp
            /// <summary>
            /// Empty if <see cref="ArgumentsCore.HasHeader"/> is false, no header presents, or upon load
            /// there was no header stored in the model.
            /// </summary>
```

---
In reply to: [247682367](https://github.com/dotnet/machinelearning/pull/2140#discussion_r247682367) [](ancestors = 247682367)

**Comment by wschin** [comment]
@TomFinley, Thank you! Let me open an issue for cleaning `TextLoader.ColInfo` because some utility functions rely on that type so removing may incur a mid-scale re-factorization.

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs [hunk 3]
```diff
@@ -520,26 +520,26 @@ public static ColInfo Create(string name, PrimitiveType itemType, Segment[] segs
...
-            public readonly Dictionary<string, int> NameToInfoIndex;
+            /// <summary>
+            /// <see cref="Infos"/>[i] stores the i-th column's metadata, named <see cref="MetadataUtils.Kinds.SlotNames"/>
+            /// in <see cref="Schema.Metadata"/>.
+            /// </summary>
             private readonly VBuffer<ReadOnlyMemory<char>>[] _slotNames;
-            // Empty iff either header+ not set in args, or if no header present, or upon load
-            // there was no header stored in the model.
+            /// <summary>
+            /// Empty if <see cref="ArgumentsCore.HasHeader"/> is false, no header presents, or upon load
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs]: >false [](start = 66, length = 5)

<see langward="false" />

File: src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs [hunk 4]
```diff
@@ -520,26 +520,26 @@ public static ColInfo Create(string name, PrimitiveType itemType, Segment[] segs
...
 
-            private readonly MetadataUtils.MetadataGetter<VBuffer<ReadOnlyMemory<char>>> _getSlotNames;
-
-            public Schema AsSchema { get; }
-
-            private Bindings()
-            {
-                _getSlotNames = GetSlotNames;
-            }
+            public Schema OutputSchema { get; }
```
Comment (sfilipi) [inline src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs]: >OutputSchema [](start = 26, length = 12)

we'd want comments on anything public.

**Pull request merged**